actioncable/lib/action_cable/connection/base.rb

```diff
@@ -67,12 +68,15 @@ def initialize(server, env, coder: ActiveSupport::JSON)
       # Called by the server when a new WebSocket connection is established. This configures the callbacks intended for overwriting by the user.
       # This method should not be called directly -- instead rely upon on the #connect (and #disconnect) callbacks.
       def process #:nodoc:
-        logger.info started_request_message
+        instrument :start_request
 
-        if websocket.possible? && allow_request_origin?
-          respond_to_successful_request
-        else
-          respond_to_invalid_request
+        instrument :upgrade_request do |payload|
+          if payload[:succeeded] = websocket.possible? && allow_request_origin?
```

Comment:
How about this for better readability? 
```suggestion
          payload[:succeeded] = websocket.possible? 
          if payload[:succeeded] && allow_request_origin?
```